app/stores/slices/api/amm.ts

```diff
@@ -35,7 +35,12 @@ export const createAmmSlice: CustomStateCreator<AmmSlice> = (set, _get) => ({
   amm: {
     ...getDefaultData(),
 
-    setAmmInfo: (amm: Amm) => {
+    setAmmInfo: (amm: Amm | string) => {
+      if (typeof amm === "string") {
+        console.error(amm);
+        return;
+      }
```

Comment:
I don't fully understand the meaning behind this portion of the code. Is this to take care of the case where the API doesn't return a valid `Amm`, and instead just says that it doesn't exist? If so, this is something that should never happen in our UI anyway once we have the right data. And even if you wanted to just make sure that it's properly taken care of anyway, we would have to do for every other slice coming from the backend, right?

Reply:
The reason why I'm doing this is because right now when we are attempting to subscribe to invalid amms, then an error message is what we get here, and I thought it would be useful to show that in console for the time being while we are integrating every bits and pieces together. Otherwise it would not be shown anywhere that we are attempting to subscribe invalid amms.